### PR TITLE
Update revoke connections feature

### PR DESCRIPTION
**Pull request by mnewlive**
Tests in the next PR

### CODE REVIEW

File: app/src/main/kotlin/com/saltedge/authenticator/features/settings/list/SettingsListViewModel.kt [hunk 1]
```diff
@@ -25,30 +25,18 @@ import android.content.DialogInterface
...
 import com.saltedge.authenticator.tools.postUnitEvent
 
 class SettingsListViewModel(
     private val appContext: Context,
-    private val appTools: AppToolsAbs,
-    private val keyStoreManager: KeyManagerAbs,
-    private val apiManager: AuthenticatorApiManagerAbs,
-    private val connectionsRepository: ConnectionsRepositoryAbs,
-    private val preferenceRepository: PreferenceRepositoryAbs
+    private val interactor: SettingsListInteractorAbs
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/features/settings/list/SettingsListViewModel.kt:39]: `SettingsListViewModel` should has possibility to interact with both APIs in the same time

File: app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt [hunk 2]
```diff
@@ -27,48 +27,26 @@ import androidx.lifecycle.*
...
-import kotlinx.coroutines.*
-import kotlin.coroutines.CoroutineContext
 
 class ConnectionsListViewModel(
     private val appContext: Context,
-    private val connectionsRepository: ConnectionsRepositoryAbs,
-    private val keyStoreManager: KeyManagerAbs,
-    private val apiManager: AuthenticatorApiManagerAbs,
-    private val cryptoTools: CryptoToolsV1Abs
+    private val interactor: ConnectionsListInteractorAbs
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt:44]: `ConnectionsListViewModel` should has possibility to interact with both APIs in the same time

File: app/src/main/kotlin/com/saltedge/authenticator/features/connections/common/ConnectionItemViewModel.kt [hunk 3]
```diff
@@ -31,5 +31,9 @@ data class ConnectionItemViewModel(
     var statusDescription: String,
     var statusDescriptionColorRes: ResId,
     val isActive: Boolean,
-    var isChecked: Boolean
-)
+    var isChecked: Boolean,
+    val apiVersion: String
+) {
+    val isV2Api: Boolean
+        get() = apiVersion == "2"
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/features/connections/common/ConnectionItemViewModel.kt:38]: core constant `API_V2_VERSION`

File: app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListInteractorV2.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    private val connectionsRepository: ConnectionsRepositoryAbs,
+    private val keyStoreManager: KeyManagerAbs,
+    private val cryptoTools: CryptoToolsV2Abs
+): ConnectionsRevokeListener {
+
+    var contract: ConnectionsListInteractorCallback? = null
+
+    override fun onConnectionsRevokeResult(apiError: ApiErrorData?) {}
+
+    fun getConnection(guid: String) {
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListInteractorV2.kt:45]: wrong name. not corresponds to function scope

File: app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt [hunk 5]
```diff
@@ -92,68 +76,94 @@ class ConnectionsListViewModel(
...
+        updateListItemEvent.postValue(listItem)
+    }
+
     fun onDeleteItemResult(guid: GUID) {
         val listItem = listItemsValues.find { it.guid == guid } ?: return
-        connectionsRepository.getByGuid(listItem.guid)?.let { connection ->
-            sendRevokeRequestForConnections(listOf(connection))
-        }
-        deleteConnectionsAndKeys(listItem.guid)
+        interactorV1.sendRevokeRequestForConnections(listItem.guid)
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListViewModel.kt:163]: check api version before `sendRevokeRequestForConnections`.  You should not send request to both APIs

File: app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListInteractorV2.kt [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    }
+
+    fun sendRevokeRequestForConnections(guid: String) {
+        connectionsRepository.getByGuid(guid)?.let { connection ->
+            sendRevokeRequestForConnections(listOf(connection))
+        }
+    }
+
+    fun collectAllConnectionsViewModels(): List<Connection> {
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListInteractorV2.kt:70]: Why you collect all connections indifferent of API version

Comment (mnewlive) [reply app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListInteractorV2.kt:70]: discussed.
Added TODO.

**Pull request merged**